Classes/Controller/Slot/Extensionmanager/ProcessActions/UpdateConfigurationFileController.php

```diff
@@ -54,32 +51,12 @@ public function updateConfigurationFile($ajaxParams, $ajaxObject)
         if (empty($extensionKey)) {
             return;
         }
-        $packageManager = Bootstrap::getInstance()->getEarlyInstance(PackageManager::class);
-        $packageManager->scanAvailablePackages();
-        $extensionConfigurationPath = $packageManager->getPackage($extensionKey)->getPackagePath() . 'ext_emconf.php';
-        $EM_CONF = null;
-        if (file_exists($extensionConfigurationPath)) {
-            $_EXTKEY = $extensionKey;
-            include $extensionConfigurationPath;
-        }
-
-        if ($EM_CONF === null || empty($EM_CONF[$extensionKey])) {
-            return;
-        }
 
-        $currentMd5HashArray = ExtensionUtility::getMd5HashArrayForExtension($extensionKey);
-        $EM_CONF[$extensionKey]['_md5_values_when_last_written'] = serialize($currentMd5HashArray);
+        $extensionUtility = GeneralUtility::makeInstance(ExtensionUtility::class);
+        $updated = $extensionUtility->updateConfiguration($extensionKey);
 
-        $emConfUtility = GeneralUtility::makeInstance(EmConfUtility::class);
-        $extensionData = [
-            'extKey' => $extensionKey,
-            'EM_CONF' => $EM_CONF[$extensionKey],
-        ];
-        $emConfContent = $emConfUtility->constructEmConf($extensionData);
-        GeneralUtility::writeFile($extensionConfigurationPath, $emConfContent);
-
-        $ajaxObject->setContentFormat('json');
         $ajaxObject->addContent('title', $this->translate('title'));
-        $ajaxObject->addContent('message', sprintf($this->translate('message'), $extensionKey));
+        $ajaxObject->addContent('message', sprintf($this->translate('message', $updated ? null : 'error'), $extensionKey));
```

Comment:
i would prefer something more readable here.
$message = $updated ? $this->translate('message') : $this->translate('error');
$ajaxObject->addContent('message', sprintf($message, $extensionKey));

Reply:
You're right. It's not very readable and in fact, that's not what it read - the condition was to switch between 1 or 2 arguments. I reverted to original flow with no output if nothing was changed.

If it should give nagative feedback then we should probably add another translation key with the same prefix?